### PR TITLE
completed typescript-destructuring exercise.

### PR DESCRIPTION
**Pull request by Katelyn-Ngo**
Screen Shot 2024-06-11 at 11 00 57 AM

### CODE REVIEW

File: typescript-destructuring/main.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+interface Book {
+  title?: string;
+  author?: string;
+  libraryId?: number;
+}
```
Comment (shawnkost) [inline typescript-destructuring/main.ts]: Are these properties going to be optional?

File: typescript-destructuring/main.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+const book1 = {
+  title: 'Goodnight Punpun',
+  author: 'Inio Asano',
+  libraryId: 3353,
+};
```
Comment (shawnkost) [inline typescript-destructuring/main.ts]: What should this object be annotated as?

File: typescript-destructuring/main.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+const { title, author, libraryId } = book1;
+console.log(
+  `The title of the book is ${title}, the author is ${author}, and the library id is ${libraryId}`
+);
+
+const book2 = {
+  title: 'Les Fleurs du mal',
+  author: 'Charles Baudelaire',
+  libraryId: 2345,
+};
```
Comment (shawnkost) [inline typescript-destructuring/main.ts]: What should this object be annotated as?

**Review by shawnkost — CHANGES_REQUESTED** [review]
Looks great Katelyn! Just a couple small typing things

**Review by brandon-moy — APPROVED** [review]
Approved! 👍

**Pull request merged**